client/subscription-edit-page/index.tsx

```diff
@@ -0,0 +1,280 @@
+			name={ inputName }
+			value={ value }
+			onChange={ ( event ) =>
+				onChange( parseInt( event.target.value, 10 ) )
+			}
+		>
+			{ options }
+		</select>
+	);
+};
+
+/**
+ * Setup the payment method select for a given element.
+ *
+ * @param {HTMLSpanElement} element The <span> where the payment method select should be rendered.
+ * @param {UserTokenCache} cache The cache of user tokens.
+ * @return {void}
+ */
+const setupPaymentSelector = (
+	element: HTMLSpanElement,
+	cache: UserTokenCache
+): void => {
+	const data = JSON.parse(
+		element.getAttribute( 'data-wcpay-pm-selector' ) || '{}'
+	) as WCPayPMSelectorData;
+
+	// Use the values from the data instead of input to ensure correct types.
+	let userId = data.userId ?? 0;
+	let value = data.value ?? 0;
+
+	// Initial population.
+	if ( userId ) {
+		cache.add( userId, data.tokens ?? [] );
+	}
+
+	// In older Subscriptions versions, there was just a simple input.
+	const input = element.querySelector( 'select,input' ) as
+		| HTMLSelectElement
+		| HTMLInputElement
+		| null;
+	if ( ! input ) {
+		return;
+	}
+
+	const root = createRoot( element );
+	const render = () => {
+		root.render(
+			<PaymentMethodSelect
+				inputName={ input.name }
+				value={ value }
+				userId={ userId }
+				cache={ cache }
+				onChange={ ( newValue: number ) => {
+					value = newValue;
+					render();
+				} }
+			/>
+		);
+	};
+
+	render();
+	cache.subscribe( render );
+	addCustomerSelectListener( async ( newUserId ) => {
+		// Once the customer is changed, the selected payment method is lost.
+		value = 0;
+		userId = newUserId;
+		render();
+
+		// Looaded, loading, or errored out, we do not need to load anything.
+		if ( cache.hasEntry( userId ) ) {
+			return;
+		}
+
+		cache.startLoading( userId );
+
+		try {
+			const response = await fetchUserTokens(
+				userId,
+				data.ajaxUrl,
+				data.nonce
+			);
+			if ( undefined === response ) {
+				throw new Error(
+					__(
+						'Failed to fetch user tokens. Please reload the page and try again.',
+						'woocommerce-payments'
+					)
+				);
+			}
+			cache.tokensLoaded( userId, response.tokens );
+		} catch ( error ) {
+			cache.loadingFailed(
+				userId,
+				error instanceof Error
+					? error.message
+					: __( 'Unknown error', 'woocommerce-payments' )
+			);
+		}
+	} );
+};
+
+/**
+ * Initializes all payment method dropdown elements on the page.
+ *
+ * Creates a shared cache for user tokens and sets up payment method
+ * selectors for all elements with the .wcpay-subscription-payment-method class.
+ *
+ * @return {void}
+ */
+const addPaymentMethodDropdowns = (): void => {
+	// Use a centralized cache for user tokens.
+	const cache = new UserTokenCache();
+
+	// There should be a single element on the page, but still make sure to iterate over all of them.
+	document
+		.querySelectorAll( '.wcpay-subscription-payment-method' )
+		.forEach( ( element ) => {
+			setupPaymentSelector( element as HTMLSpanElement, cache );
+		} );
```

Comment:
(minor) to remove the forced casting 

```suggestion
	document
		.querySelectorAll< HTMLSpanElement >(
			'.wcpay-subscription-payment-method'
		)
		.forEach( ( element ) => {
			setupPaymentSelector( element, cache );
		} );
```

Reply:
Addressed in https://github.com/Automattic/woocommerce-payments/pull/11143/commits/4043b4914c6f4e20531c154d2396f124212d7fe1.